test-harness/helpers.ts

```diff
@@ -96,19 +96,22 @@ export function tmpFile(opts?: tmp.TmpNameOptions): Promise<tmp.FileResult> {
   });
 }
 
-const BRACES = /{([^}]+)}/g;
+const BRACES = /{[^}]+}/g;
 
 export const applyReplacements = (str: string, values: Dictionary<string>) => {
-  BRACES.lastIndex = 0;
-  let result: RegExpExecArray | null;
-
-  // tslint:disable-next-line:no-conditional-assignment
-  while ((result = BRACES.exec(str))) {
-    if (!(result[1] in values)) continue;
-    const newValue = String(values[result[1]] || '');
-    str = `${str.slice(0, result.index)}${newValue}${str.slice(BRACES.lastIndex)}`;
-    BRACES.lastIndex = result.index + newValue.length;
-  }
+  const replacer = (match: string): string => {
+    const identifier = match.slice(1, -1);
```

Comment:
We could keep the capture group to avoid the slice here. Groups are passed to replacer, so you can easily access them.

Reply:
@P0lip Thanks! Fixed.